x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -213,17 +209,30 @@ export class SessionIndex {
         'Attempted to create a new session while session index is initializing.'
       );
       await this.indexInitialization;
+    } else {
+      await this.ensureSessionIndexExists();
     }
 
+    // **************************************************
```

Comment:
__question:__ Since we no longer use the index template, how do you feel about creating an alias for the session index and using [`require_alias`](https://github.com/elastic/elasticsearch/pull/58917) option for the document create operation?

If we assume that the session index exists most of the time, then with this option we could avoid calling `index.exists` for every new session since we won't need to worry about automatically created session index anymore. If `.kibana_security_session/_create/12345?require_alias=true` fails with `404 index_not_found_exception` then we can either fail and ask for a restart (like Task Manager does, if I'm not mistaken) or try to re-create index with proper settings (and optionally retry to create session document, it's not a big deal if we don't do this - users can try to to login once again).

What do you think?

Reply:
I think I like this idea! I implemented this in https://github.com/elastic/kibana/pull/134900/commits/b890529164e49a570e374318dbda7e742b91a6bd, take a look and let me know what you think. I have a few more tests to write if you like this approach

Reply:
Looks great, thanks for making this change! Let me know when it's ready for review.